ci/src/main/scala/org/typelevel/sbt/TypelevelCiPlugin.scala

```diff
@@ -46,12 +46,101 @@ object TypelevelCiPlugin extends AutoPlugin {
         cond = Some(primaryJavaCond.value)
       )
     ),
-    githubWorkflowJavaVersions := Seq(JavaSpec.temurin("8"))
+    githubWorkflowJavaVersions := Seq(JavaSpec.temurin("8")),
+    GlobalScope / Keys.onLoad := {
+      val oses = githubWorkflowOSes.value.toList
+      val javas = githubWorkflowJavaVersions.value.toList
+      val scalas = githubWorkflowScalaVersions.value.toList
+      val additions = githubWorkflowBuildMatrixAdditions.value
+      val inclusions = githubWorkflowBuildMatrixInclusions.value.toList
+      val exclusions = githubWorkflowBuildMatrixExclusions.value.toList
+      val stepPreamble = githubWorkflowBuildSbtStepPreamble.value.toList
+
+      (GlobalScope / Keys.onLoad).value.compose { (state: State) =>
+        addCiAlias(
+          state,
+          oses,
+          javas,
+          scalas,
+          additions,
+          inclusions,
+          exclusions,
+          stepPreamble,
+          githubWorkflowBuild.value
+        )
+      }
+    },
+    GlobalScope / Keys.onUnload := {
+      (GlobalScope / Keys.onUnload)
+        .value
+        .compose((state: State) => BasicCommands.removeAlias(state, "ci"))
+    }
   )
 
   private val primaryJavaCond = Def.setting {
     val java = githubWorkflowJavaVersions.value.head
     s"matrix.java == '${java.render}'"
   }
 
+  private def addCiAlias(
+      state: State,
+      oses: List[String],
+      javaVersions: List[JavaSpec],
+      scalaVersions: List[String],
+      matrixAdditions: Map[String, List[String]],
+      matrixInclusions: List[MatrixInclude],
+      matrixExclusions: List[MatrixExclude],
+      sbtStepPreamble: List[String],
+      workflowSteps: Seq[WorkflowStep]) = {
+
+    val buildMatrix = GenerativePlugin.expandMatrix(
+      // Cannot meaningfully iterate OS or Java version here
+      oses = oses.take(1),
+      javas = javaVersions.take(1),
+      scalas = scalaVersions,
+      matrixAdds = matrixAdditions,
+      includes = matrixInclusions,
+      excludes = matrixExclusions
+    )
+
+    val keys = "os" :: "scala" :: "java" :: matrixAdditions.keys.toList.sorted
+
+    val commands = for {
+      matrixRow <- buildMatrix
+      matrixValues = keys.zip(matrixRow).toMap
+      step <- workflowSteps.collect {
+        case sbt: WorkflowStep.Sbt if matchingCondition(sbt, matrixValues) => sbt
+      }
+      command <- sbtStepPreamble ++ step.commands
+    } yield replaceMatrixVars(command, matrixValues)
+
+    val commandAlias = TypelevelKernelPlugin.mkCommand(commands)
+
+    BasicCommands.addAlias(
+      state,
+      "ci",
+      commandAlias
+    )
+  }
+
+  private def matchingCondition(
+      command: WorkflowStep.Sbt,
+      matrixValues: Map[String, String]) = {
+    // For all matrix values
+    matrixValues.forall {
+      case (k, v) =>
+        val renderedCond = s"matrix.$k == '$v'"
+
+        command.cond.forall { cond =>
```

Comment:
Is it likely that the condition itself might refer to a matrix variable? Perhaps `cond` needs to have its matrix vars replaced too.

Reply:
Yes, that's very likely. In fact we do it from sbt-typelevel itself.

https://github.com/typelevel/sbt-typelevel/blob/a4282aed116872daa6b9cd77c40804ceacc73f6f/ci/src/main/scala/org/typelevel/sbt/CrossRootProject.scala#L154

Reply:
Ah that should work OK - what I was worried about was things like
```scala
Some("something == '${{ matrix.project }}'")
``` 
In that scenario we would need to make sure to replace any Github Actions `${{ }}` variables before checking the conditions.

Reply:
Hmm, I might be confused but in conditions you should assume everything is wrapped in `${{ }}`. Like, the `matrix.project` in that condition should definitely be substituted, no? :)

Reply:
https://docs.github.com/en/actions/learn-github-actions/expressions

> When you use expressions in an if conditional, you may omit the expression syntax (${{ }}) because GitHub automatically evaluates the if conditional as an expression.